jdbi3-spring:3.51.0 and newer break Spring Boot 3

## Reporting an issue to Jdbi

[X] Jdbi Version: 3.51.0 - 3.54.0<br>
[X] Jdbi module: jdbi3-spring<br>
[X] Java Version: 21.0.10-librca<br>
[X] Spring Boot 3.5.16<br>
[X] I have read and understood the [AI contribution guidelines](https://github.com/jdbi/jdbi/blob/master/AI-GUIDELINES.md)<br>
[X] I understand that by contributing code to Jdbi, it may be distributed to others under the [Apache Software License 2.0](https://github.com/jdbi/jdbi/blob/master/LICENSE). I am authorized to contribute this code.<br>

----
AI Disclosure: Claude Agent was used to bisect every release between `3.48.0` and `3.54.0` to
identify the version `spring-jcl` vanished from compile/runtimeClasspath.

This issue only appears with Spring Boot 3.x. Spring Boot 4.x works fine. Spring 7 dropped `spring-jcl`and depends directly on `commons-logging:commons-logging`.

Adding `org.jdbi:jdbi3-spring` (jdbi3-bom 3.51.0 or later) to a Gradle project that also uses Spring Boot's dependency-management version causes `org.springframework:spring-jcl` disappear from the resolved compile/runtime classpath entirely.

```
./gradlew bootRun

Exception in thread "main" java.lang.NoClassDefFoundError: org/apache/commons/logging/LogFactory
	at org.springframework.boot.SpringApplication.<clinit>(SpringApplication.java:202)
	at com.example.repro.ApplicationKt.main(Application.kt:42)
Caused by: java.lang.ClassNotFoundException: org.apache.commons.logging.LogFactory
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:188)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:526)
	... 2 more
```

Bare bones Spring Boot application that reproduces the issue.
`Application.kt`
```kotlin
package com.example.repro

import jakarta.servlet.FilterChain
import jakarta.servlet.http.HttpServletRequest
import jakarta.servlet.http.HttpServletResponse
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.runApplication
import org.springframework.boot.web.servlet.FilterRegistrationBean
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.web.filter.OncePerRequestFilter

@SpringBootApplication
class Application

fun main(args: Array<String>) {
    runApplication<Application>(*args)
}

// OncePerRequestFilter is used to trigger the issue. It's logger field needs
// org.apache.commons.logging.Log/LogFactory (spring-jcl) to be resolvable.
@Configuration
class NoOpFilterConfig {

    @Bean
    fun noOpFilterRegistration(): FilterRegistrationBean<NoOpFilter> =
        FilterRegistrationBean(NoOpFilter()).apply {
            urlPatterns = listOf("/*")
        }
}

class NoOpFilter : OncePerRequestFilter() {
    override fun doFilterInternal(request: HttpServletRequest, response: HttpServletResponse, filterChain: FilterChain) {
        filterChain.doFilter(request, response)
    }
}
```
`build.gradle.kts`
```kotlin
plugins {
    id("org.springframework.boot") version "3.5.16"
    id("io.spring.dependency-management") version "1.1.7"
    kotlin("jvm") version "2.1.20"
    kotlin("plugin.spring") version "2.1.20"
}

group = "com.example"
version = "0.0.1-SNAPSHOT"

java {
    toolchain {
        languageVersion = JavaLanguageVersion.of(21)
    }
}

repositories {
    mavenCentral()
}

dependencies {
    // Swap this to 3.50.0 to see spring-jcl reappear and app doesn't throw an exception.
    implementation(platform("org.jdbi:jdbi3-bom:3.54.0"))

    implementation("org.springframework.boot:spring-boot-starter-web")
    implementation("org.jdbi:jdbi3-spring")
}

```

--- a/RELEASE_NOTES.md
+++ b/RELEASE_NOTES.md
@@ -1,5 +1,7 @@
 # Unreleased
 
+- Fix jdbi3-spring excluding spring-jcl from consumers since 3.51.0, which broke Spring Boot 3
+  applications at startup with `NoClassDefFoundError: org.apache.commons.logging.LogFactory` (#2990)
 - Fix PreparedBatch NPE when rows bind different runtime types, e.g. mixed bean subclasses (#2974, thanks @arimu1!)
 - Fix DefaultJdbiCache pinning entries when loader throws an exception (#2995)
 - Fix GraalVM native image missing entries and update metadata to new format, support 25.2 (#2994)

--- a/internal/build/pom.xml
+++ b/internal/build/pom.xml
@@ -252,12 +252,6 @@
                 <groupId>org.springframework</groupId>
                 <artifactId>spring-core</artifactId>
                 <version>${dep.spring.version}</version>
-                <exclusions>
-                    <exclusion>
-                        <groupId>org.springframework</groupId>
-                        <artifactId>spring-jcl</artifactId>
-                    </exclusion>
-                </exclusions>
             </dependency>
 
             <dependency>

--- a/internal/build/pom.xml
+++ b/internal/build/pom.xml
@@ -254,12 +254,6 @@
                 <groupId>org.springframework</groupId>
                 <artifactId>spring-core</artifactId>
                 <version>${dep.spring.version}</version>
-                <exclusions>
-                    <exclusion>
-                        <groupId>org.springframework</groupId>
-                        <artifactId>spring-jcl</artifactId>
-                    </exclusion>
-                </exclusions>
             </dependency>
 
             <dependency>

--- a/native-tests/pom.xml
+++ b/native-tests/pom.xml
@@ -137,6 +137,14 @@
             <groupId>org.jdbi</groupId>
             <artifactId>jdbi3-spring</artifactId>
             <scope>test</scope>
+            <!-- spring-jcl duplicates the commons-logging classes that jcl-over-slf4j
+                 (via pg-embedded's maven-loader) provides on the test classpath. -->
+            <exclusions>
+                <exclusion>
+                    <groupId>org.springframework</groupId>
+                    <artifactId>spring-jcl</artifactId>
+                </exclusion>
+            </exclusions>
         </dependency>
         <dependency>
             <groupId>org.jdbi</groupId>
@@ -280,6 +288,12 @@
             <version>${project.version}</version>
             <type>test-jar</type>
             <scope>test</scope>
+            <exclusions>
+                <exclusion>
+                    <groupId>org.springframework</groupId>
+                    <artifactId>spring-jcl</artifactId>
+                </exclusion>
+            </exclusions>
         </dependency>
         <dependency>
             <groupId>org.jdbi</groupId>

--- a/spring/pom.xml
+++ b/spring/pom.xml
@@ -62,18 +62,6 @@
         </dependency>
 
         <dependency>
-            <groupId>org.slf4j</groupId>
-            <artifactId>jcl-over-slf4j</artifactId>
-            <scope>test</scope>
-        </dependency>
-
-        <dependency>
-            <groupId>org.slf4j</groupId>
-            <artifactId>slf4j-simple</artifactId>
-            <scope>test</scope>
-        </dependency>
-
-        <dependency>
             <groupId>org.jdbi</groupId>
             <artifactId>jdbi3-testing</artifactId>
             <scope>test</scope>

--- a/spring5/pom.xml
+++ b/spring5/pom.xml
@@ -74,18 +74,6 @@
         </dependency>
 
         <dependency>
-            <groupId>org.slf4j</groupId>
-            <artifactId>jcl-over-slf4j</artifactId>
-            <scope>test</scope>
-        </dependency>
-
-        <dependency>
-            <groupId>org.slf4j</groupId>
-            <artifactId>slf4j-simple</artifactId>
-            <scope>test</scope>
-        </dependency>
-
-        <dependency>
             <groupId>com.h2database</groupId>
             <artifactId>h2</artifactId>
             <scope>test</scope>

